src/categories/categories.service.ts

```diff
@@ -58,19 +64,25 @@ export class CategoriesService {
     }
   }
 
-  async getAllApis(categoryId: string) {
+  async getAllApis(
+    categoryId: string,
+    query: PaginateQuery,
+  ): Promise<Paginated<Api>> {
     try {
-      const category = await this.categoryRepo.findOne({
-        where: { id: categoryId },
-      });
-      if (category.name.toLowerCase() == 'all')
-        return await this.apiRepo.find({
-          where: { visibility: Visibility.Public },
-        });
-      const apis = await this.apiRepo.find({
-        where: { categoryId, visibility: Visibility.Public },
+      return paginate(query, this.apiRepo, {
+        sortableColumns: ['createdOn', 'name', 'visibility'],
+        searchableColumns: ['name', 'description', 'about', 'visibility'],
+        defaultSortBy: [['id', 'DESC']],
+        where: {
+          visibility: Visibility.Public,
+          categoryId: categoryId,
```

Comment:
This is not the way to use nestjs paginate package, you don't pass in the category id directly, there is a query object that's supposed to carry that data, How are you going to use categoryId param together with query? .Why don't you read the docs of the package before using it.

This endpoint always return empty array for me, if it works on your side, I'll like to see.

Reply:
I remover the condition that checks for category.name category.name === 'all' because it doesn't matter anymore since there would be a categoryId it will fetch all the apis under the categoryId specified

Reply:
Looks good, still getting empty array

Reply:
Are you sure there are apis in the category you are  testing with?

Reply:
Ofcourse, can you show me a screenshot?

Reply:
![Uploading Screenshot from 2023-02-23 08-34-05.png…]()

Reply:
![Screenshot from 2023-02-23 08-34-05](https://user-images.githubusercontent.com/38318744/220844644-27cb962f-0afe-40e5-8778-917e5f3ce730.png)
![Screenshot from 2023-02-23 08-35-18](https://user-images.githubusercontent.com/38318744/220844647-b757fcea-0c1f-4558-8a2e-5f35c62f74e0.png)